xmodule/mongo_utils.py

```diff
@@ -30,8 +30,11 @@ def connect_to_mongodb(
     handles AutoReconnect errors by retrying read operations, since these exceptions
     typically indicate a temporary step-down condition for MongoDB.
     """
-    # If the MongoDB server uses a separate authentication database that should be specified here
-    auth_source = kwargs.get('authsource', '') or None
+    # If the MongoDB server uses a separate authentication database that should be specified here.
+    # Convert the lowercased authsource parameter to the camel-cased authSource expected by MongoClient.
+    auth_source = db
+    if auth_source_key := {'authSource', 'authsource'}.intersection(set(kwargs.keys())):
+        auth_source = kwargs.pop(auth_source_key)
```

Comment:
It needs to be
```
auth_source = kwargs.pop(auth_source_key.pop())
```
Because auth_source_key is a set now.

Reply:
Good catch, thanks. Fixed.